watcher.go

```diff
@@ -0,0 +1,171 @@
+package main
+
+import (
+	"fmt"
+	"github.com/fsnotify/fsnotify"
+	pollingWatcher "github.com/radovskyb/watcher"
+	"log"
+	"os"
+	"path/filepath"
+	"regexp"
+	"syscall"
+	"time"
+)
+
+type FileWatcher interface {
+	Close() error
+	AddFiles(pattern *regexp.Regexp) error
+	add(path string) error
+	Watch(jobs chan<- string, pattern *regexp.Regexp)
+}
+
+type NotifyWatcher struct {
+	watcher *fsnotify.Watcher
+}
+
+func (n NotifyWatcher) Close() error {
+	return n.watcher.Close()
+}
+
+func (n NotifyWatcher) AddFiles(pattern *regexp.Regexp) error {
+	return addFiles(n)
+}
+
+func (n NotifyWatcher) Watch(jobs chan<- string, pattern *regexp.Regexp) {
+	for {
+		select {
+		case ev := <-n.watcher.Events:
+			if ev.Op&fsnotify.Remove == fsnotify.Remove || ev.Op&fsnotify.Write == fsnotify.Write || ev.Op&fsnotify.Create == fsnotify.Create {
+				base := filepath.Base(ev.Name)
+
+				// Assume it is a directory and track it.
+				if *flagRecursive == true && !flagExcludedDirs.Matches(ev.Name) {
+					n.watcher.Add(ev.Name)
+				}
+
+				if flagIncludedFiles.Matches(base) || matchesPattern(pattern, ev.Name) {
+					if !flagExcludedFiles.Matches(base) {
```

Comment:
This is duplicated code. Can you refactor it to something like `pathMatches(ev.Name, pattern)` something like

```
func pathMatches(ev.Name string, pattern *regexp.Regexp)  {
    base := filepath.Base(ev.Name)
    return (flagIncludedFiles.Matches(base) || matchesPattern(pattern, ev.Name)) &&
					!flagExcludedFiles.Matches(base)
}
```

Reply:
👍